ethcore/src/engines/null_engine.rs

```diff
@@ -64,4 +69,42 @@ impl Engine for NullEngine {
 	fn snapshot_components(&self) -> Option<Box<::snapshot::SnapshotComponents>> {
 		Some(Box::new(::snapshot::PowSnapshot(10000)))
 	}
+
+	fn on_close_block(&self, block: &mut ExecutedBlock) -> Result<CloseOutcome, Error> {
+		if !self.params.apply_reward {
+			return Ok(CloseOutcome{trace: None});
+		}
+
+		/// Block reward
+		let fields = block.fields_mut();		
+		let mut tracer = ExecutiveTracer::default();
+
+		let result_block_reward = U256::from(1000000000);
+		fields.state.add_balance(
+			fields.header.author(),
+			&result_block_reward,
+			CleanupMode::NoEmpty
+		)?;
+
+		let block_miner = fields.header.author().clone();
+		tracer.trace_reward(block_miner, result_block_reward, RewardType::Block);
+
+		/// Uncle rewards
+		let result_uncle_reward = U256::from(10000000);
+		for u in fields.uncles.iter() {
+			let uncle_miner = u.author().clone();
+			fields.state.add_balance(
+				u.author(),
+				&(result_uncle_reward),
+				CleanupMode::NoEmpty
+			)?;
+			tracer.trace_reward(uncle_miner, result_uncle_reward, RewardType::Uncle);
+		}
+
+		fields.state.commit()?;
+		match *fields.tracing_enabled {
```

Comment:
I think it's because `Executive` operates only on state and transaction not on full blocks.
IMHO it would be cleaner to actually alter the traces directly here.